hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson

```diff
@@ -204,4 +204,163 @@
       tests: ["otp_ctrl_stress_all"]
     }
   ]
+
+  covergroups: [
+    {
+      name: power_on_cg
+      desc: '''Covers the following conditions when OTP_CTRL finishes power-on initialization:
+            - whether `lc_escalation_en` is On
+            - whether any partition (except life cycle partition) is locked
+            '''
+    }
+    {
+      name: flash_req_cg
+      desc: '''Covers whether secret1 partition is locked during `flash` data or address
+            request.'''
+    }
+    {
+      name: sram_req_cg
+      desc: '''Covers whether secret1 partition is locked during all `srams` key request.'''
+    }
+    {
+      name: otbn_req_cg
+      desc: '''Covers whether secret1 partition is locked during `otbn` key request.'''
+    }
+    {
+      name: lc_prog_cg
+      desc: '''Covers whether the error bit is set during LC program request.'''
+    }
+    {
+      name: keymgr_o_cg
+      desc: '''Covers the following conditions when scoreboard checks `keymgr_o` value:
+            - whether secret2 partition is locked
+            - whether `lc_seed_hw_rd_en_i` is On
+            '''
+    }
+    {
+      name: req_dai_access_after_alert_cg
+      desc: '''Covers if sequence issued various DAI requests after any fatal alert is
+            triggered.'''
+    }
+    {
+      name: issue_checks_after_alert_cg
+      desc: '''Covers if sequence issued various OTP_CTRL's background checks after any fatal alert
+            is triggered.'''
+    }
+    { // TODO: this is from a wrapper class, confirm if this is the correct format
+      name: csr_rd_after_alert_cg
+      desc: '''Covers if the following CSRs are being read and the value is checked in scoreboard
+            after any fatal alert is triggered:
+            - unbuffered partitions' digest CSRs
+            - HW partition' digest CSRs
+            - secrets partitions' digest CSRs
+            - direct_access read data CSRs
+            - status CSR
+            - error_code CSR
+            '''
+    }
+    {
+      name: dai_err_code_cg
+      desc: '''Covers all applicable error codes in DAI, and cross each error code with all
+               7 partitions.'''
+    }
+    {
+      name: lci_err_code_cg
+      desc: '''Covers all applicable error codes in LCI.'''
+    }
+    { // TODO: this is an array of covergroups, confirm if this is the correct format
+      name: unbuf_err_code_cg[0]
```

Comment:
@sriyerg do you think we can add another entry called `size` to support array of covergroups?
It might look like this:
```
    {
      name: unbuf_err_code_cg
      size: 2
      desc: '''Covers all applicable error codes in all unbuffered partitions.'''
    }
```